src/IO/LasLoader/lasloader.cpp

```diff
@@ -0,0 +1,237 @@
+    double offset_x = *( reinterpret_cast<double*>( buffer + 24 ) );
+    double offset_y = *( reinterpret_cast<double*>( buffer + 32 ) );
+    double offset_z = *( reinterpret_cast<double*>( buffer + 40 ) );
+
+    /*loading properties*/
+    vector<char> point( data_len );
+
+    VectorArray<Ra::Core::Vector3>& vertices = geometry->getGeometry().verticesWithLock();
+    vertices.reserve( nb_data );
+
+    // checking for colors
+    Ra::Core::Utils::AttribHandle<Ra::Core::Vector4> handle_color;
+    Ra::Core::VectorArray<Ra::Core::Vector4> colors;
+    if ( data_format == 2 || data_format == 3 || ( minor >= 3 && data_format == 5 ) ) {
+        handle_color = geometry->getGeometry().vertexAttribs().addAttrib<Ra::Core::Vector4>(
+            Ra::Core::Geometry::getAttribName( Ra::Core::Geometry::MeshAttrib::VERTEX_COLOR ) );
+        LOG( logINFO ) << "\"red\", \"green\", \"blue\".";
+    }
+
+    // checking for GPS Time
+    Ra::Core::Utils::AttribHandle<Scalar> handle_time;
+    Ra::Core::VectorArray<Scalar> gps_time;
+    if ( ( data_format == 1 ) || ( data_format == 3 ) ||
+         ( ( minor > 2 ) && ( data_format == 4 ) ) || ( ( minor > 2 ) && ( data_format == 5 ) ) ||
+         ( ( minor == 4 ) && ( data_format == 6 ) ) ) {
+        handle_time = geometry->getGeometry().vertexAttribs().addAttrib<Scalar>( "gps_time" );
+        LOG( logINFO ) << "\"GPS time\".";
+    }
+
+    for ( unsigned int i = 0; i < nb_data; ++i ) {
+        // positioning at point data location in file
+        int pos = offset + i * data_len;
+        stream.seekg( 0 );
+        stream.seekg( pos );
+
+        // reading point data
+        stream.read( point.data(), data_len );
+
+        // extracting initial x y z coordinates
+        int ix;
+        int iy;
+        int iz;
+
+        ix = *(int*)( point.data() );
+        iy = *(int*)( point.data() + 4 );
+        iz = *(int*)( point.data() + 8 );
+
+        // computing actual coordinates
+        double x = (double)ix * scale_x + offset_x;
+        double y = (double)iy * scale_y + offset_y;
+        double z = (double)iz * scale_z + offset_z;
+
+        // storing coordinates in vertices object
+        vertices.emplace_back( Scalar( x ), Scalar( y ), Scalar( z ) );
+
+        // loading colors if found
+        if ( handle_color.idx().isValid() ) {
+            unsigned short red, green, blue;
+
+            if ( data_format == 5 || data_format == 3 ) {
+                red   = *(unsigned short*)( point.data() + 28 );
+                green = *(unsigned short*)( point.data() + 30 );
+                blue  = *(unsigned short*)( point.data() + 32 );
+            }
+            else {
+                red   = *(unsigned short*)( point.data() + 20 );
+                green = *(unsigned short*)( point.data() + 22 );
+                blue  = *(unsigned short*)( point.data() + 24 );
+            }
+
+            colors.emplace_back( Scalar( red ), Scalar( green ), Scalar( blue ), 1_ra );
```

Comment:
As you can see here https://storm-irit.github.io/Radium-Engine/release-candidate/classRa_1_1Core_1_1Utils_1_1ColorBase.html
Colors have values between 0 and 1. 
As you cast unsigned short to float, you will always have values > 1.

Reply:
so do I need to perform a division by 2¹⁶?

Reply:
You need to be sure what is the maximum value in practice. Usually it is 255.

Reply:
it says in the specification that color channels are coded on 16 bits and that it should always be normalized to be so